### PR TITLE
feat(TDP-5419): implement business requirements

### PR DESCRIPTION
**Pull request by frassinier**
**Link to the JIRA issue**
https://jira.talendforge.org/browse/TDP-5419

**Please check if the PR fulfills these requirements**
- [x] The commit(s) message(s) follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md#commit-message-format)
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [x] The code coverage on new code is > 75 % for backend and > 95% for frontend
- [x] The new code does not introduce new technical issues (sonar / eslint)
- [x] Functional tests have been performed
- [ ] Docker configuration files for config-std or config-cloud profiles are impacted

**Please check the browsers you've tested on**
- [x] Chrome, Firefox, Safari, Edge, IE11
- [ ] No, that's bad, this PR should not be merged !
- [ ] No, and no need to (backend changes only)

### CODE REVIEW

File: dataprep-api/src/main/java/org/talend/dataprep/api/service/settings/actions/provider/InventoryActions.java [hunk 1]
```diff
@@ -28,6 +29,7 @@
             .name("inventory.edit")
             .icon("talend-pencil")
             .type("@@inventory/EDIT")
+            .data(DATA_FEATURE_KEY, "inventory.rename")
```
Comment (ncomont) [inline dataprep-api/src/main/java/org/talend/dataprep/api/service/settings/actions/provider/InventoryActions.java]: Should we also track the rename action from the preparation view ? 
image

File: dataprep-api/src/main/java/org/talend/dataprep/api/service/settings/actions/provider/MenuActions.java [hunk 2]
```diff
@@ -37,6 +35,7 @@
             .name("menu.preparations")
             .icon("talend-dataprep")
             .type("@@router/GO_CURRENT_FOLDER")
+            .data(DATA_FEATURE_KEY, "preparation.open")
```
Comment (ncomont) [inline dataprep-api/src/main/java/org/talend/dataprep/api/service/settings/actions/provider/MenuActions.java]: I fear that the double click on the table name won't work.

File: dataprep-webapp/src/app/components/transformation/menu/transformation-menu.html [hunk 3]
```diff
@@ -23,6 +23,7 @@
     <li ng-repeat="menu in menuCtrl.menuItems track by menu.name">
         <a href
            title="{{ menu.label }}"
+           data-feature="{{ 'column.' + menu.name }}"
```
Comment (ncomont) [inline dataprep-webapp/src/app/components/transformation/menu/transformation-menu.html]: Rename is also possible by double clicking on the column header. Should we track it too ?

File: dataprep-webapp/src/app/components/export/export.html [hunk 4]
```diff
@@ -50,6 +50,7 @@
                     class="btn btn-default talend-modal-close modal-secondary-button"
                     translate-once="CANCEL"></button>
             <input class="btn btn-primary talend-modal-close modal-primary-button"
+                   data-feature="preparation.export.start"
```
Comment (ncomont) [inline dataprep-webapp/src/app/components/export/export.html]: What about the Download button in the full run history page ?

File: dataprep-webapp/src/app/components/widgets-containers/inventory-list/inventory-list-controller.js [hunk 5]
```diff
@@ -220,21 +225,24 @@ export default class InventoryListCtrl {
...
+		return this.SettingsActionsService.adaptDataAttributes(itemAction);
 	}
 
 	createDropdownActions(items, actionName) {
 		return items.map((item) => {
 			const itemAction = this.createDropdownItemAction(item, actionName);
 			itemAction.label = item.label || item.name;
-			return itemAction;
+			if (item.locationType) {
+				itemAction['data-feature'] = `dataset.${item.locationType}.add`;
```
Comment (ncomont) [inline dataprep-webapp/src/app/components/widgets-containers/inventory-list/inventory-list-controller.js]: Should we also track the "Import file" option of the "Add preparation" modal ?

File: dataprep-api/src/main/java/org/talend/dataprep/api/service/settings/actions/provider/MenuActions.java (hunk 2 above)
Comment (frassinier) [reply dataprep-api/src/main/java/org/talend/dataprep/api/service/settings/actions/provider/MenuActions.java]: Our analytics solution only tracks click AFAIK

**Pull request merged**